spec/cc/workspace/path_tree_benchmark_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "spec_helper"
+
+module CC::Workspace
+  describe PathTree do
+    include FileSystemHelpers
+
+    MAX_FILTER_SECONDS = 10
+    NUM_FILES = 10_000
+    AVG_FILES_PER_DIR = 20
+    REPO_DEPTH = 30
+    NUM_EXCLUDES = 50
+
+    it "performs acceptably" do
```

Comment:
I vote we put this and other benchmarks under `./benchmarks`, with a separate rake task. I'd also vote they output timing information (e.g. `Benchmark.bm`), rather than assert on timings being under some threshold.

I wouldn't characterize what happened as developers just not realizing, because nothing was watching their backs. Rather, I think we would've actively used benchmarks during the development of this logic, had we had them -- and we probably should've written them then. Therefore, I think benchmarks being available to run and view as needed is just as valuable as putting performance-based assertions in the normal spec suite.

Maybe some day we could incorporate it into the release process, to run the benchmarks between master and the last released tag.